dataikuapi/dss/ml.py

```diff
@@ -308,50 +334,757 @@ def set_metric(self, metric=None, custom_metric=None, custom_metric_greater_is_b
+        if stratified is not None:
+            if not isinstance(stratified, bool):
+                warnings.warn("HyperparameterSearchSettings.set_validation_mode_to_single_split ignoring invalid input: stratified")
+                warnings.warn("stratified must be a boolean")
+            else:
+                self._raw_settings["stratified"] = stratified
+
+    def set_validation_mode_to_custom(self, code=None):
+        self._raw_settings["mode"] = "CUSTOM"
+        if code is not None:
+            if not is_basestring(code):
+                warnings.warn("HyperparameterSearchSettings.set_validation_mode_to_custom ignoring invalid input: code")
+                warnings.warn("code must be a Python interpretable string")
+            self._raw_settings["code"] = code
+
+    def set_search_distribution(self, distributed, n_containers=None):
+        assert isinstance(distributed, bool)
+        if n_containers is not None:
+            assert isinstance(n_containers, int)
+            self._raw_settings["nContainers"] = n_containers
+        self._raw_settings["distributed"] = distributed
+
+    @property
+    def timeout(self):
+        return self._raw_settings["timeout"]
+
+    @timeout.setter
+    def timeout(self, timeout):
+        assert isinstance(timeout, int)
+        self._raw_settings["timeout"] = timeout
+
+    @property
+    def n_iter(self):
+        if self._raw_settings["strategy"] == "GRID":
+            return self._raw_settings["nIter"]
+        else:
+            # RANDOM and BAYESIAN search strategies
+            return self._raw_settings["nIterRandom"]
+
+    @n_iter.setter
+    def n_iter(self, n_iter):
+        assert isinstance(n_iter, int)
+        if self._raw_settings["strategy"] == "GRID":
+            self._raw_settings["nIter"] = n_iter
+        else:
+            self._raw_settings["nIterRandom"] = n_iter
+
+    @property
+    def parallelism(self):
+        return self._raw_settings["nJobs"]
+
+    @parallelism.setter
+    def parallelism(self, n_jobs):
+        self._raw_settings["nJobs"] = n_jobs
+
+
+class HyperparameterSettings(object):
+
+    def __init__(self, name, algo_settings):
+        self.name = name
+        self._algo_settings = algo_settings
+
+
+class NumericalHyperparameterSettings(HyperparameterSettings):
+
+    def _repr_html_(self):
+        active_dict = self._algo_settings._get_active_numerical_hyperparam_dict(self.name)
+        return "<pre>" + self.__class__.__name__ + "(hyperparameter=\"{}\", settings={})</pre>".format(self.name, json.dumps(active_dict, indent=4)) + \
+               "\n<details><pre>{}</pre></details>".format(self.__repr__())
+
+    def __repr__(self):
+        raw_dict = self._algo_settings[self.name]
+        return self.__class__.__name__ + "(hyperparameter=\"{}\", settings={})".format(self.name, json.dumps(raw_dict, indent=4))
+
+    __str__ = __repr__
+
+    def set_mode_to_explicit(self, values=None):
```

Comment:
[Personal opinion] it feels weird, when I am already in "EXPLICIT" and want to change values, to have to use a method called `set_mode_to_explicit`

Reply:
In the same vein as the new method names for `HyperparameterSearchSettings` (`set_grid_search`...), we could name this one `set_explicit_mode` etc. I guess this would sound less weird ?

Reply:
`set_explicit_values(self, values)`?